File: examples/files/native_modules/Props.m
#import <Foundation/Foundation.h>

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(MyViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(myInt, Int)
RCT_EXPORT_VIEW_PROPERTY(myColor, UIColor)
RCT_EXPORT_VIEW_PROPERTY(myArray, NSArray)

@end

File: examples/files/native_modules/Props.swift
class MyView: UIView {

  @objc var myInt: Int = 0 {
    didSet {
      // Do something...
    }
  }

  @objc var myColor: UIColor = UIColor.black {
    didSet {
      // Do something...
    }
  }

  @objc var myArray: NSArray = [] {
    didSet {
      // Do something...
    }
  }

}